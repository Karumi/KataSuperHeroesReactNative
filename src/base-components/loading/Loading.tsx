import * as React from "react";
import { ActivityIndicator, StyleSheet } from "react-native";

class Loading extends React.Component {
    public render() {
        return <ActivityIndicator size="large" color="white" style={style.activityIndicator} />;
    }
}

const style = StyleSheet.create({
    activityIndicator: {
        position: "absolute",
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
        alignItems: "center",
        justifyContent: "center",
    },
});

export default Loading;
