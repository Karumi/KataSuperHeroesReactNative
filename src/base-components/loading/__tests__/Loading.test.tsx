import * as React from "react";
import * as renderer from "react-test-renderer";

import Loading from "../Loading";

it("renders with the default configuration", () => {
    const component = renderer
        .create(<Loading />)
        .toJSON();
    expect(component).toMatchSnapshot();
});
