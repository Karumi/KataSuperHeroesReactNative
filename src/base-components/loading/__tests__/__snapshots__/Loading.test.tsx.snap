// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the default configuration 1`] = `
<ActivityIndicator
  animating={true}
  color="white"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;
